src/main/kotlin/uk/gov/justice/digital/hmpps/breachnoticeapi/listener/DomainEventsListener.kt

```diff
@@ -0,0 +1,105 @@
+package uk.gov.justice.digital.hmpps.breachnoticeapi.listener
+import com.fasterxml.jackson.annotation.JsonAnyGetter
+import com.fasterxml.jackson.annotation.JsonAnySetter
+import com.fasterxml.jackson.annotation.JsonProperty
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.module.kotlin.readValue
+import io.awspring.cloud.sqs.annotation.SqsListener
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+import uk.gov.justice.digital.hmpps.breachnoticeapi.entity.BreachNoticeEntity
+import uk.gov.justice.digital.hmpps.breachnoticeapi.enums.ReviewEventType
+import uk.gov.justice.digital.hmpps.breachnoticeapi.service.BreachNoticeService
+import uk.gov.justice.digital.hmpps.breachnoticeapi.service.NDeliusIntegrationService
+import java.time.LocalDateTime
+
+@Service
+class DomainEventsListener(
+  private val breachNoticeService: BreachNoticeService,
+  private val objectMapper: ObjectMapper,
+  private val nDeliusIntegrationService: NDeliusIntegrationService,
+) {
+
+  @Transactional
+  @SqsListener("hmppsbreachnoticequeue", factory = "hmppsQueueContainerFactoryProxy")
+  fun listen(msg: String) {
+    val (message, attributes) = objectMapper.readValue<SQSMessage>(msg)
+    val domainEventMessage = objectMapper.readValue<DomainEventsMessage>(message)
+    handleMessage(domainEventMessage)
+  }
+
+  private fun handleMessage(message: DomainEventsMessage) {
+    when (message.eventType) {
+      "probation-case.merge.completed" -> {
+        // Update CRNs where appropriate
+        val breachNotices = breachNoticeService.getActiveBreachNoticesForCrn(message.sourceCrn)
+        breachNotices.forEach {
+          message.targetCrn?.let { it1 -> breachNoticeService.updateBreachNoticeCrn(it, it1) }
+        }
+
+        updateReviewEvent(ReviewEventType.MERGE, breachNotices, message.occurredAt)
+      }
+
+      "probation-case.unmerge.completed" -> {
+        // Update CRNs where appropriate
+        val breachNotices = breachNoticeService.getActiveBreachNoticesForCrn(message.unmergedCrn)
+        breachNotices.forEach {
+          nDeliusIntegrationService.getCrnForBreachNoticeUuid(it.id.toString())?.crn?.let { it1 ->
+            breachNoticeService.updateBreachNoticeCrn(
+              it,
+              it1,
+            )
+          }
```

Comment:
Could we use a better name than `it1` 😄 

```suggestion
          nDeliusIntegrationService.getCrnForBreachNoticeUuid(it.id.toString())?.crn?.let { crn ->
            breachNoticeService.updateBreachNoticeCrn(it, crn)
          }
```